internal/serialization/generic_compact.go

```diff
@@ -77,55 +80,65 @@ func (cm GenericCompact) Read(reader serialization.CompactReader) interface{} {
-	//FieldKindNullableInt16          FieldKind = 37
-	//FieldKindArrayOfNullableInt16   FieldKind = 38
-	//FieldKindNullableInt32          FieldKind = 39
-	//FieldKindArrayOfNullableInt32   FieldKind = 40
-	//FieldKindNullableInt64          FieldKind = 41
-	//FieldKindArrayOfNullableInt64   FieldKind = 42
-	//FieldKindNullableFloat32        FieldKind = 43
-	//FieldKindArrayOfNullableFloat32 FieldKind = 44
-	//FieldKindNullableFloat64        FieldKind = 45
-	//FieldKindArrayOfNullableFloat64 FieldKind = 46
+var fieldKindToType map[serialization.FieldKind]reflect.Type
 
+func init() {
+	var a any
+	var b bool
+	var i8 int8
+	var i16 int16
+	var i32 int32
+	var i64 int64
+	var f32 float32
+	var f64 float64
+	var t time.Time
+	fieldKindToType = map[serialization.FieldKind]reflect.Type{
+		serialization.FieldKindNotAvailable:                 nil,
+		serialization.FieldKindBoolean:                      reflect.TypeOf(b),
+		serialization.FieldKindArrayOfBoolean:               reflect.TypeOf([]bool{}),
+		serialization.FieldKindInt8:                         reflect.TypeOf(i8),
+		serialization.FieldKindArrayOfInt8:                  reflect.TypeOf([]int8{}),
+		serialization.FieldKindInt16:                        reflect.TypeOf(i16),
+		serialization.FieldKindArrayOfInt16:                 reflect.TypeOf([]int16{}),
+		serialization.FieldKindInt32:                        reflect.TypeOf(i32),
+		serialization.FieldKindArrayOfInt32:                 reflect.TypeOf([]int32{}),
+		serialization.FieldKindInt64:                        reflect.TypeOf(i64),
+		serialization.FieldKindArrayOfInt64:                 reflect.TypeOf([]int64{}),
+		serialization.FieldKindFloat32:                      reflect.TypeOf(f32),
+		serialization.FieldKindArrayOfFloat32:               reflect.TypeOf([]float32{}),
+		serialization.FieldKindFloat64:                      reflect.TypeOf(f64),
+		serialization.FieldKindArrayOfFloat64:               reflect.TypeOf([]float64{}),
+		serialization.FieldKindString:                       reflect.TypeOf(""),
+		serialization.FieldKindArrayOfString:                reflect.TypeOf([]string{}),
+		serialization.FieldKindDecimal:                      reflect.TypeOf(types.NewDecimal(new(big.Int), 0)),
+		serialization.FieldKindArrayOfDecimal:               reflect.TypeOf([]types.Decimal{}),
+		serialization.FieldKindTime:                         reflect.TypeOf(types.LocalTime(t)),
+		serialization.FieldKindArrayOfTime:                  reflect.TypeOf([]types.LocalTime{}),
+		serialization.FieldKindDate:                         reflect.TypeOf(types.LocalTime(t)),
+		serialization.FieldKindArrayOfDate:                  reflect.TypeOf([]types.LocalDate{}),
+		serialization.FieldKindTimestamp:                    reflect.TypeOf(types.LocalTime(t)),
+		serialization.FieldKindArrayOfTimestamp:             reflect.TypeOf([]types.LocalDateTime{}),
+		serialization.FieldKindTimestampWithTimezone:        reflect.TypeOf(types.LocalTime(t)),
```

Comment:
I think there is small mistake. It could be "types.OffsetDateTime(t)"?

Reply:
Thanks, fixed at: https://github.com/hazelcast/hazelcast-commandline-client/pull/159/commits/ff8e67dd3673dbb32a0005c0509f3b81dc9944a4